coins/monero/src/transaction.rs

```diff
@@ -214,12 +216,18 @@ pub struct Transaction {
 impl Transaction {
   pub(crate) fn fee_weight(
     protocol: Protocol,
+    decoy_weights: &Vec<usize>,
     inputs: usize,
     outputs: usize,
     extra: usize,
+    fee: u64,
   ) -> usize {
-    TransactionPrefix::fee_weight(protocol.ring_len(), inputs, outputs, extra) +
-      RctSignatures::fee_weight(protocol, inputs, outputs)
+    if inputs != decoy_weights.len() {
```

Comment:
Replace inputs with decoy_weights.len()? Why take in the variable twice if it just introduces a potential inconsistency?

Reply:
> Why take in the variable twice if it just introduces a potential inconsistency?

Because I didn't think of this here, good spot :)